### ISSUE TITLE
Explore File compression

### ISSUE THREAD

**Issue by nakedsushi** [ProjectLoudly/planning#24]
See if it's worth it or feasible to do at least some sort of file compressing on the client side so that we're not trying to upload a giant uncompressed file.

The two tactics we can use with compression could be:

1. Chunk the file and compress each chunk before uploading.  Server is responsible for decompressing the chunks and stitching them together. Server then encodes that wave to flac.

2. Compress the file and then chunk it. Server is responsible for stitching the files together and then decompressing it. Server then encodes that wave to flac.

JS libraries to check out:
https://github.com/nmrugg/LZMA-JS
https://blog.slava.online/xz-archiver-webworker-for-browser-pure-js/

AC:
- [ ] Client can upload a compressed file
- [ ] Server produces the correct FLAC file.

**Comment by nakedsushi** [comment]
Welp, I implemented a POC of client-side file compression and what we learned is that client-side file compression is not going to work for us.

1.) Memory constraints -- for large files, compressing them is really memory heavy and the browser tends to crash because the compression ends up using > 1Gb of memory.

2.) Speed constraints -- it takes about 4x as long to just compress a file as it does to upload the uncompressed file.

Mostly because of 2, I don't think it's worth trying to solve 1, since the point of exploring file compression is to speed up upload speed.  I've searched around for alternative compression algorithms, but most of the audio-specific ones have to do with encoding, which is something we decided we didn't want to do in a previous sprint b/c of the memory issue.  For now, I think we've spent enough cycles going down this rabbit hole and have at least have an educated answer about why we're giving it up.
